tests/integration/test_interface.py

```diff
@@ -15,29 +15,64 @@
 )
 from pseudopeople.schema_entities import COLUMNS, FORMS
 
+# TODO: Move into a metadata file and import metadata into prl
+DATA_COLUMNS = ["year", "event_date", "survey_date", "tax_year"]
+
 
 @pytest.mark.parametrize(
-    "data_dir_name, noising_function",
+    "data_dir_name, noising_function, use_sample_data",
     [
-        ("decennial_census_observer", generate_decennial_census),
-        ("household_survey_observer_acs", generate_american_communities_survey),
-        ("household_survey_observer_cps", generate_current_population_survey),
-        ("social_security_observer", generate_social_security),
-        ("tax_w2_observer", generate_taxes_w2_and_1099),
-        ("wic_observer", generate_women_infants_and_children),
-        ("tax 1040", "todo"),
+        ("decennial_census_observer", generate_decennial_census, True),
+        ("decennial_census_observer", generate_decennial_census, False),
+        ("household_survey_observer_acs", generate_american_communities_survey, True),
+        ("household_survey_observer_acs", generate_american_communities_survey, False),
+        ("household_survey_observer_cps", generate_current_population_survey, True),
+        ("household_survey_observer_cps", generate_current_population_survey, False),
+        ("social_security_observer", generate_social_security, True),
+        ("social_security_observer", generate_social_security, False),
+        ("tax_w2_observer", generate_taxes_w2_and_1099, True),
+        ("tax_w2_observer", generate_taxes_w2_and_1099, False),
+        ("wic_observer", generate_women_infants_and_children, True),
+        ("wic_observer", generate_women_infants_and_children, False),
+        ("tax 1040", "todo", True),
+        ("tax 1040", "todo", False),
     ],
 )
-def test_generate_form(data_dir_name: str, noising_function: Callable):
+def test_generate_form(
+    data_dir_name: str, noising_function: Callable, use_sample_data: bool, tmpdir
+):
+    """Tests that noised forms are generated and as expected. The 'use_sample_data'
+    parameter determines whether or not to use the sample data (if True) or
+    a non-default root directory with multiple datasets to compile (if False)
+    """
     if noising_function == "todo":
         pytest.skip(reason=f"TODO: implement form {data_dir_name}")
-    # todo fix hard-coding in MIC-3960
-    data_path = paths.SAMPLE_DATA_ROOT / data_dir_name / f"{data_dir_name}.parquet"
-    data = pd.read_parquet(data_path)
 
-    noised_data = noising_function(seed=0)
-    noised_data_same_seed = noising_function(seed=0)
-    noised_data_different_seed = noising_function(seed=1)
+    sample_data_path = list(
+        (paths.SAMPLE_DATA_ROOT / data_dir_name).glob(f"{data_dir_name}*")
+    )[0]
```

Comment:
Why are you only grabbing the first file?

Reply:
We have control over the sample data and know that there is only one dataset per form